packages/browser-lite/src/database.ts

```diff
@@ -0,0 +1,552 @@
+    direction: SortDirection;
+}
+
+export interface FilterOption<T extends Record> {
+    field: Extract<keyof T, string>;
+    kind: FilterKind;
+    value: any;
+}
+
+export enum SortDirection {
+    Asc = 'ASC',
+    Desc = 'DESC',
+}
+
+export enum FilterKind {
+    Equal = '=',
+    NotEqual = '!=',
+    Less = '<',
+    Greater = '>',
+    LessOrEqual = '<=',
+    GreaterOrEqual = '>=',
+    Contains = 'CONTAINS',
+}
+
+export interface Order {
+    hash: string;
+    chainId: number;
+    makerAddress: string;
+    makerAssetData: string;
+    makerAssetAmount: string;
+    makerFee: string;
+    makerFeeAssetData: string;
+    takerAddress: string;
+    takerAssetData: string;
+    takerFeeAssetData: string;
+    takerAssetAmount: string;
+    takerFee: string;
+    senderAddress: string;
+    feeRecipientAddress: string;
+    expirationTimeSeconds: string;
+    salt: string;
+    signature: string;
+    exchangeAddress: string;
+    fillableTakerAssetAmount: string;
+    lastUpdated: string;
+    isRemoved: number;
+    isPinned: number;
+    parsedMakerAssetData: string;
+    parsedMakerFeeAssetData: string;
+}
+
+export type OrderField = keyof Order;
+
+export type OrderQuery = Query<Order>;
+
+export type OrderSort = SortOption<Order>;
+
+export type OrderFilter = FilterOption<Order>;
+
+export interface AddOrdersResult {
+    added: Order[];
+    removed: Order[];
+}
+
+export interface MiniHeader {
+    hash: string;
+    parent: string;
+    number: string;
+    timestamp: string;
+    logs: string;
+}
+
+export type MiniHeaderField = keyof MiniHeader;
+
+export type MiniHeaderQuery = Query<MiniHeader>;
+
+export type MiniHeaderSort = SortOption<MiniHeader>;
+
+export type MiniHeaderFilter = FilterOption<MiniHeader>;
+
+export interface AddMiniHeadersResult {
+    added: MiniHeader[];
+    removed: MiniHeader[];
+}
+
+export interface Metadata {
+    ethereumChainID: number;
+    maxExpirationTime: string;
+    ethRPCRequestsSentInCurrentUTCDay: number;
+    startOfCurrentUTCDay: string;
+}
+
+function newNotFoundError(): Error {
+    return new Error('could not find existing model or row in database');
+}
+
+function newMetadataAlreadExistsError(): Error {
+    return new Error('metadata already exists in the database (use UpdateMetadata instead?)');
+}
+
+/**
+ * Creates and returns a new database
+ *
+ * @param opts The options to use for the database
+ */
+export function createDatabase(opts: Options): Database {
+    return new Database(opts);
+}
+
+export class Database {
+    private readonly _db: Dexie;
+    // private readonly maxOrders: number;
+    private readonly _maxMiniHeaders: number;
+    private readonly _orders: Dexie.Table<Order, string>;
+    private readonly _miniHeaders: Dexie.Table<MiniHeader, string>;
+    private readonly _metadata: Dexie.Table<Metadata, number>;
+
+    constructor(opts: Options) {
+        this._db = new Dexie(opts.dataSourceName);
+        // this._maxOrders = opts.maxOrders;
```

Comment:
I think you mean "why was this commented"?

Max orders and dynamic max expiration time are not implemented right now. It's one of the main things that still needs to be done in a follow-up PR. Uncommenting it produces a warning because `_maxOrders` is never read.

Reply:
Yes I did -- my bad. Thanks for the explanation.